app/session.go

```diff
@@ -113,6 +121,18 @@ func (a *App) RevokeAllSessions(userId string) *model.AppError {
 	return nil
 }
 
+// RevokeSessionsFromAllUsers will go through all the sessions active
+// in the server and revoke them
+func (a *App) RevokeSessionsFromAllUsers() *model.AppError {
+	err := a.Srv.Store.Session().RemoveAllSessions()
+	if err != nil {
+		return err
+	}
+	a.ClearSessionCacheForAllUsers()
```

Comment:
This should be much better from a performance and cluster impact point of view.

One question though - not sure if you've thought bout this already, but in the individual case (your `revokeSession()` method) we have a distinction between revoking sessions and revoking access tokens. Do we need to any such special handling here?

Reply:
it shouldn't be special at all, I'll add some more code to revoke tokens as well, someone with an access token and ill intentions would probably re-create the sessions in no time which would be problematic. Also, revoking 1 user and N should have the same effects.
Thanks for pointing this!